tests/bgp/test_route_aggregation.py

```diff
@@ -0,0 +1,187 @@
+          -c "network 1.1.1.0/24 route-map AGG_TEST_1" -c "network 1.1.2.0/24 route-map AGG_TEST_1" \
+          -c "network 1.1.3.0/24 route-map AGG_TEST_2" -c "network 1.1.4.0/24 route-map AGG_TEST_2" \
+          -c "network 1.1.5.0/24" -c "address-family ipv6 unicast" -c "network 1:1:1:1::/64 route-map AGG_TEST_1" \
+          -c "network 1:1:1:2::/64 route-map AGG_TEST_1" -c "network 1:1:1:3::/64 route-map AGG_TEST_2" \
+          -c "network 1:1:1:4::/64 route-map AGG_TEST_2" -c "network 1:1:1:5::/64 route-map AGG_TEST_2"'.format(
+              gather_info['cli_options'], gather_info['dut_asn'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "5", False)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure summary-only route aggregation
+    agg_configuration(True, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "summary-only", "summary-only")
+    logger.debug(gather_info['duthost'].shell('vtysh -c "show run bgp"', module_ignore_errors=True)['stdout'])
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "1", False)
+
+    # Remove the aggregate configuration
+    agg_configuration(False, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "summary-only", "summary-only")
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "5", False)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure as-set summary-only route aggregation
+    agg_configuration(True, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "as-set summary-only", "as-set summary-only")
+    logger.debug(gather_info['duthost'].shell('vtysh -c "show run bgp"', module_ignore_errors=True)['stdout'])
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "1", False,
+                     as_set=True)
+
+    # Remove the aggregate configuration
+    agg_configuration(False, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "as-set summary-only", "as-set summary-only")
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "5", False)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure route aggregation with suppress-map
+    # Create prefix lists to be used
```

Comment:
Same here, could we encapsulate these configuration details into a separate function to keep the main routine clean and readable?

Reply:
They already are encapsulated.